mm2src/lp_bot/simple_market_maker.rs

```diff
@@ -0,0 +1,266 @@
+    #[display(fmt = "The bot is already started")]
+    AlreadyStarted,
+    #[display(fmt = "Invalid bot configuration")]
+    InvalidBotConfiguration,
+    #[display(fmt = "Transport error: {}", _0)]
+    Transport(String),
+    #[display(fmt = "Internal error: {}", _0)]
+    InternalError(String),
+}
+
+#[derive(Debug)]
+pub enum PriceServiceRequestError {
+    HttpProcessError(String),
+}
+
+impl From<std::string::String> for PriceServiceRequestError {
+    fn from(error: String) -> Self { PriceServiceRequestError::HttpProcessError(error) }
+}
+
+impl From<std::str::Utf8Error> for PriceServiceRequestError {
+    fn from(error: Utf8Error) -> Self { PriceServiceRequestError::HttpProcessError(error.to_string()) }
+}
+
+impl HttpStatusCode for StartSimpleMakerBotError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            StartSimpleMakerBotError::AlreadyStarted | StartSimpleMakerBotError::InvalidBotConfiguration => {
+                StatusCode::BAD_REQUEST
+            },
+            StartSimpleMakerBotError::Transport(_) | StartSimpleMakerBotError::InternalError(_) => {
+                StatusCode::INTERNAL_SERVER_ERROR
+            },
+        }
+    }
+}
+
+impl HttpStatusCode for StopSimpleMakerBotError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            // maybe bad request is not adapted for the first errors.
+            StopSimpleMakerBotError::AlreadyStopped | StopSimpleMakerBotError::AlreadyStopping => {
+                StatusCode::BAD_REQUEST
+            },
+            StopSimpleMakerBotError::Transport(_) | StopSimpleMakerBotError::InternalError(_) => {
+                StatusCode::INTERNAL_SERVER_ERROR
+            },
+        }
+    }
+}
+
+pub async fn tear_down_bot(ctx: MmArc) {
+    let simple_market_maker_bot_ctx = TradingBotContext::from_ctx(&ctx).unwrap();
+    {
+        let mut trading_bot_cfg = simple_market_maker_bot_ctx.trading_bot_cfg.lock().await;
+        // todo: check if clear is adapted, if i understand its keep the memory allocated for later usage.
+        trading_bot_cfg.clear();
+    }
+    // todo: cancel all pending orders
+}
+
+async fn update_single_order(_cfg: SimpleCoinMarketMakerCfg, uuid: Uuid, _order: MakerOrder, key_trade_pair: String) {
+    info!("need to update order: {} of {}", uuid, key_trade_pair)
+}
+
+async fn process_bot_logic(ctx: &MmArc) {
+    let simple_market_maker_bot_ctx = TradingBotContext::from_ctx(ctx).unwrap();
+    // note: Copy the cfg here will not be expensive, and this will be thread safe.
+    let cfg = simple_market_maker_bot_ctx.trading_bot_cfg.lock().await.clone();
+
+    let mut memoization_pair_registry: HashSet<String> = HashSet::new();
+    let maker_orders = retrieve_my_maker_orders(ctx).await;
+
+    info!("np_orders: {}", maker_orders.len());
+    for (key, value) in maker_orders.into_iter() {
+        let key_trade_pair = value.base.clone() + "/" + value.rel.clone().as_str();
```

Comment:
I think that the `TradingPair` struct will look better here. It can also be a key for the `HashMap`.
```rust
TradingPair {
    base: String,
    rel: String,
}
```

Reply:
Here the JSON configuration may look like that:

```json
"cfg": {
            "KMD-BEP20/BUSD-BEP20": {
                "base": "KMD-BEP20",
                "rel": "BUSD-BEP20",
                "max": true,
                "min_volume": "0.25",
                "spread": "1.025",
                "base_confs": 3,
                "base_nota": false,
                "rel_confs": 1,
                "rel_nota": false,
                "enable": true
            }
        }
  ```
  
  If I do the implementation above, I would need an extra helper function to return it as `BASE/REL` to access the key from the configuration which is fine!

Reply:
should be solved in https://github.com/KomodoPlatform/atomicDEX-API/pull/1066/commits/9e6fa2703ba862b6cd8439b936d5d2acf6a4e4d0